src/pages/Timetable.tsx

```
@@ -19,13 +22,62 @@ const TimetablePage: React.FC = () => {
+    if (searchQueries.courseName) params.courseName = searchQueries.courseName;
+
+    setApiParams(params);
+  }, [filters, searchQueries, currentPage]);
+
+  // Use fallback data if API is not available
+  const displayData = data && data.length > 0 ? data : COURSE_DATA;
+  const displayTotalPages = totalPages > 0 ? totalPages : MOCK_API_RESPONSE.totalPages;
 
   const navigate = useNavigate();
-  const goDetail = (subjectCode: string) => navigate(subjectCode);
+  const goDetail = (courseNumber: string, courseData?: any) => {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)